### DISCUSSION TITLE
Dynamically Creating 'Enums'

### DISCUSSION THREAD

**Discussion by aarondglover** [alirezanet/InfiniteEnumFlags#6] [Q&A]
This library looks awesome and I'm hooked on Bitwise for permissions etc. The issue of exceeding 64 is annoying to overcome so I am interested in applying this to my project. Thumbs up to the author 👍

I should not be lazy and just prototype it... But.... I am lazy so I figured I'd ask and maybe it will help someone else too

**My question is whether an Infinite Enum could be dynamically generated at run time?** 

Imagine creating an app which has the ability to add dynamic roles. Could an infiniteEnum/flag be created... Generate and store that unique I'd against the role.

Then when  I assign that dynamic role to a user, get the uniqueId, convert it back - then assign that enum to a user's set of permissions using bitwise... Then generate and persist a new UniqueId.

**Comment by alirezanet** [comment]
Hi @aarondglover, 
Interesting Idea, I haven't thought about it before but technically yes it is possible. using the `var dynamicFlag = new Flag<YourEnumType>(index);` you can create it dynamically, where the `index` must be your last created flag +1.  but it really depends on what kind of problem you want to solve and how. 

I suggest you try it, and let me know if you faced any problems.
